apps/AppWithWearable/lib/utils/ble/scan.dart

```diff
@@ -39,6 +39,7 @@ Future<BTDeviceStruct?> scanAndConnectDevice({bool autoConnect = true}) async {
 
 Future<List<BTDeviceStruct?>> scanDevices() async {
   List<BTDeviceStruct> foundDevices = await bleFindDevices();
-  var filteredDevices = foundDevices.where((device) => device.name == "Friend").toList();
+  // var filteredDevices = foundDevices.where((device) => device.name == "Friend").toList();
```

Comment:
https://github.com/BasedHardware/Friend/pull/276#issuecomment-2171751353

Reply:
see here for more discussion https://github.com/BasedHardware/Friend/pull/225#issuecomment-2169764690

Reply:
This would show all ble available devices right? as far as I understand, if both options work, why to do it like this is what I don't understand

Reply:
No it will not show all BLE devices. If the device is named different from "Friend" it will not show up. When that line is active it is filtering BLE devices by name. It will go through the onboarding screen with a real device as long as its name is "Friend" but not with a simulator. Recently we stopped filtering devices by name and use UUID instead. So this line is not only unnecessary but also prevents us from pairing a simulator.

See https://github.com/BasedHardware/Friend/pull/225#issuecomment-2169764690